src/Lean/Parser/Tactic.lean

```diff
@@ -125,6 +125,17 @@ example : 1 + 1 = 2 := by rfl
 @[builtin_tactic_parser] def decide := leading_parser
   nonReservedSymbol "decide"
 
+/--
+`decide!` is a variant of the `decide` tactic that uses kernel reduction to prove the goal.
+It has the following properties:
+- Since it uses kernel reduction instead of elaboratior reduction, it ignores transparancy and can unfold everything.
+- While `decide` needs to reduce the `Decidable` instance twice (once during elaboration to verify whether the tactic succeeds,
+  and once during kernel type checking), the `decide!` tactic reduces it exactly once.
+- When `decide!` is successful, the result is cached for the current module.
```

Comment:
Maybe remove that from the docstring. We don’t want people to relying on that behavior if we don't know if and how that will work post-parallelism.
```suggestion
```

Reply:
Sure. How does leaving the caching test sound, so we know when this finally has broken?

Reply:
Absolutely!